src/scvi/external/methylvi/_model.py

```diff
@@ -0,0 +1,627 @@
+    """
+    Model class for methylVI :cite:p:`Weinberger2023a`
+
+    Parameters
+    ----------
+    mdata
+        MuData object that has been registered via :meth:`~scvi.external.METHYLVI.setup_mudata`.
+    n_hidden
+        Number of nodes per hidden layer.
+    n_latent
+        Dimensionality of the latent space.
+    n_layers
+        Number of hidden layers used for encoder and decoder NNs.
+    **model_kwargs
+        Keyword args for :class:`~scvi.external.methylvi.METHYLVAE`
+
+    Examples
+    --------
+    >>> mdata = mudata.read_h5mu(path_to_mudata)
+    >>> MethylVI.setup_mudata(mdata, batch_key="batch")
+    >>> vae = MethylVI(mdata)
+    >>> vae.train()
+    >>> mdata.obsm["X_methylVI"] = vae.get_latent_representation()
+    """
+
+    def __init__(
+        self,
+        mdata: MuData,
+        n_hidden: int = 128,
+        n_latent: int = 10,
+        n_layers: int = 1,
+        **model_kwargs,
+    ):
+        super().__init__(mdata)
+
+        n_batch = self.summary_stats.n_batch
+
+        # We feed in both the number of methylated counts (mc) and the
+        # total number of counts (cov) as inputs
+        self.modalities = self.get_anndata_manager(mdata).modalities
+
+        self.num_features_per_modality = [mdata[modality].shape[1] for modality in self.modalities]
+
+        n_input = np.sum(self.num_features_per_modality)
+
+        self.module = METHYLVAE(
+            n_input=n_input,
+            n_hidden=n_hidden,
+            n_latent=n_latent,
+            n_layers=n_layers,
+            n_batch=n_batch,
+            modalities=self.modalities,
+            num_features_per_modality=self.num_features_per_modality,
+            **model_kwargs,
+        )
+        self._model_summary_string = (
+            "Overwrite this attribute to get an informative representation for your model"
+        )
+        # necessary line to get params that will be used for saving/loading
+        self.init_params_ = self._get_init_params(locals())
+
+        logger.info("The model has been initialized")
+
+    @classmethod
+    @setup_anndata_dsp.dedent
+    def setup_anndata(
+        cls,
+        adata: AnnData,
+        **kwargs,
+    ) -> AnnData | None:
+        """
+        %(summary)s.
+
+        Parameters
+        ----------
+        %(param_mdata)s
+        mc_layer
+            Layer containing methylated cytosine counts for each set of methylation features.
+        cov_layer
+            Layer containing total coverage counts for each set of methylation features.
+        %(param_batch_key)s
+
+        Returns
+        -------
+        %(returns)s
+        """
+        raise NotImplementedError("METHYLVI must be used with a MuData object.")
+
+    @classmethod
+    @setup_anndata_dsp.dedent
+    def setup_mudata(
+        cls,
+        mdata: MuData,
+        mc_layer: str,
+        cov_layer: str,
+        methylation_modalities: Iterable[str],
+        batch_key: str | None = None,
```

Comment:
Can you add support for categorical covariates? We will deprecate batch_key soon'ish as it's confusing for users (difference between batch_key and categorical_covariate_key).

Reply:
Done!

Reply:
I'm missing it?

Reply:
If you check one line below here I've added the `categorical_covariate_keys` parameter